[JIT] Move jit functional tests creation to testing utils

eellison opened the pull request:
This moves our creation of functional tests to internal utils. I plan on using it internally for benchmarking. There are no functional changes. 

ghstack-source-id: 6314cc80d0c133cc3d144490cb812085dbf3b7b8
Pull Request resolved: https://github.com/pytorch/pytorch/pull/34694

driazati reviewed (approved):
Looks fine but can you put it into it's own file instead of adding to `jit_utils`? Something like `torch/testing/_internal/tests/functional.py`

Commit:
Move functional graph creation to testing utils

ghstack-source-id: 56c64dcd841e45999ba032468f619a2f7a9010e9
Pull Request resolved: https://github.com/pytorch/pytorch/pull/34694
